### PR TITLE
Feature/change http client

### PR DESCRIPTION
**Pull request by mateusluizfb**
Issue: https://github.com/prodis/correios-cep/issues/15

This PR change the HTTP client from `NET::HTTP` to `http.rb`

### CODE REVIEW

File: lib/correios/cep/web_service.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # frozen_string_literal: true
-require 'net/https'
+# require 'net/https'
```
Comment (prodis) [inline lib/correios/cep/web_service.rb]: Code commented leftover.

File: lib/correios/cep/web_service.rb [hunk 2]
```diff
@@ -23,43 +24,12 @@ def initialize
...
-        )
-      end
-
-      def build_request(zipcode)
-        request = Net::HTTP::Post.new(uri.path)
-        request['Content-Type'] = CONTENT_TYPE_HEADER
-        request.body = BODY_TEMPLATE % { zipcode: zipcode }
-        request
+        HTTP
+          .use(logging: {logger: Logger.new(STDOUT)})
```
Comment (prodis) [inline lib/correios/cep/web_service.rb]: Logger configuration is not being used.

File: lib/correios/cep/web_service.rb [hunk 3]
```diff
@@ -23,43 +24,12 @@ def initialize
...
+          .headers('Content-Type' => CONTENT_TYPE_HEADER)
+          .post(@uri.to_s, body: BODY_TEMPLATE % { zipcode: zipcode })
```
Comment (prodis) [inline lib/correios/cep/web_service.rb]: Please extract a method that generates the body.

File: lib/correios/cep/web_service.rb [hunk 4]
```diff
@@ -23,43 +24,12 @@ def initialize
...
-
-      private
-
-      attr_reader :uri, :proxy_uri
-
-      def build_http
-        Net::HTTP.start(
-          uri.host,
-          uri.port,
-          proxy_uri.host,
```
Comment (prodis) [inline lib/correios/cep/web_service.rb]: Proxy configuration were ignored.

File: lib/correios/cep/web_service.rb [hunk 5]
```diff
@@ -23,43 +24,12 @@ def initialize
...
-          proxy_uri.port,
-          nil,
-          nil,
-          use_ssl: true,
-          verify_mode: OpenSSL::SSL::VERIFY_NONE,
```
Comment (prodis) [inline lib/correios/cep/web_service.rb]: Please check if it is necessary to avoid SSL verification.

File: lib/correios/cep/web_service.rb [hunk 6]
```diff
@@ -23,43 +24,12 @@ def initialize
...
-          open_timeout: Correios::CEP.request_timeout,
```
Comment (prodis) [inline lib/correios/cep/web_service.rb]: Timeout configuration was ignored.

File: lib/correios/cep/web_service.rb (hunk 2 above)
Comment (mateusluizfb) [reply lib/correios/cep/web_service.rb]: What do you mean? I tested here and when the request is made and received the log prints

File: lib/correios/cep/web_service.rb (hunk 3 above)
Comment (mateusluizfb) [reply lib/correios/cep/web_service.rb]: You mean all the code between row 27 and 30 or just the `.body.to_s`?

File: lib/correios/cep/web_service.rb (hunk 2 above)
Comment (prodis) [reply lib/correios/cep/web_service.rb]: You should use the logger is set on configuration. For example: https://github.com/prodis/correios-cep#configuration-example

In your implementation you fixed the logger to STDOUT.

File: lib/correios/cep/web_service.rb (hunk 3 above)
Comment (prodis) [reply lib/correios/cep/web_service.rb]: I mean:
```ruby
BODY_TEMPLATE % { zipcode: zipcode }
```

**Comment by prodis** [comment]
@mateusluizfb Any news? 😄

**Comment by mateusluizfb** [comment]
Sorry, I'm a little busy with work and college, but I didn't forget!! Tomorrow I'll try to see ;)

**Pull request merged**